app/va/vetext/client.py

```diff
@@ -29,72 +31,81 @@ def init_app(
         self.logger = logger
         self.statsd = statsd
 
+    @staticmethod
+    def format_for_vetext(payload: V2PushPayload) -> dict[str, str]:
+        if payload.personalisation is not None:
+            payload.personalisation = {f'%{k.upper()}%': v for k, v in payload.personalisation.items()}
+
+        if payload.is_broadcast():
+            formatted_payload = {
+                'appSid': payload.app_sid,
+                'templateSid': payload.template_id,
+                'topicSid': payload.topic_sid,
+                'personalization': payload.personalisation,
+            }
+        else:
+            formatted_payload = {
+                'appSid': payload.app_sid,
+                'icn': payload.icn,
+                'templateSid': payload.template_id,
+                'personalization': payload.personalisation,
+            }
```

Comment:
nit: have a base formatted_payload and just add in what is needed depending on if/else

Reply:
Updated.